src/lib.rs

```diff
@@ -1721,13 +1776,7 @@ pub trait Itertools : Iterator {
         where Self: Sized + Iterator<Item = T::Item>,
               T: traits::HomogeneousTuple
     {
-        match self.next_tuple() {
-            elt @ Some(_) => match self.next() {
-                Some(_) => None,
-                None => elt,
-            },
-            _ => None
-        }
+        self.next_tuple().filter(|_| self.next().is_none())
```

Comment:
Is this really relevant to this PR? If not, could we separate it into another PR?

Reply:
It was mostly to be consistent with the other implementation. As it's just a stylistic change I don't think it's worth a pull request by itself to be honest.